tabledesc: omit implicit columns from auto-generated index names

otan opened the pull request:
Release note (sql change): Omit the inclusion of implicit columns or
hash sharded index columns from automatically generated index names.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/58898)
<!-- Reviewable:end -->

otan committed:
tabledesc: omit implicit columns from auto-generated index names

Release note (sql change): Omit the inclusion of implicit columns or
hash sharded index columns from automatically generated index names.

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/58898#-:-MQtqqK9AcJMnhd5yz1s:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm

otan commented:
bors r=ajstorm
